code/datums/components/item_two_handed.dm

```diff
@@ -0,0 +1,339 @@
+		src.force_unwielded = force_unwielded
+	if(icon_prefix)
+		src.icon_prefix = icon_prefix
+	if(icon_update_callback)
+		src.icon_update_callback = icon_update_callback
+	if(on_wield_callback)
+		src.on_wield_callback = on_wield_callback
+	if(on_unwield_callback)
+		src.on_unwield_callback = on_unwield_callback
+
+// register signals withthe parent item
+/datum/component/two_handed/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, .proc/on_equip)
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/on_drop)
+	RegisterSignal(parent, COMSIG_ITEM_ATTACK_SELF, .proc/on_attack_self)
+	RegisterSignal(parent, COMSIG_ITEM_ATTACK, .proc/on_attack)
+	RegisterSignal(parent, COMSIG_ATOM_UPDATE_ICON, .proc/on_update_icon)
+	RegisterSignal(parent, COMSIG_TRY_TWOHANDED_WIELD, .proc/try_wield)
+	RegisterSignal(parent, COMSIG_TRY_TWOHANDED_UNWIELD, .proc/try_unwield)
+
+// Remove all siginals registered to the parent item
+/datum/component/two_handed/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_ITEM_EQUIPPED,
+								COMSIG_ITEM_DROPPED,
+								COMSIG_ITEM_ATTACK_SELF,
+								COMSIG_ATOM_UPDATE_ICON,
+								COMSIG_TRY_TWOHANDED_WIELD,
+								COMSIG_TRY_TWOHANDED_UNWIELD))
+
+/// Triggered on equip of the item containing the component
+/datum/component/two_handed/proc/on_equip(datum/source, mob/user, slot)
+	if(require_twohands && slot == ITEM_SLOT_HANDS) // force equip the item
+		wield(user)
+	if(!user.is_holding(parent) && wielded && !require_twohands)
+		unwield(user)
+
+/// Triggered on drop of item containing the component
+/datum/component/two_handed/proc/on_drop(datum/source, mob/user)
+	if(require_twohands)
+		unwield(user, show_message=TRUE)
+	if(wielded)
+		unwield(user)
+
+/// Triggered on attack self of the item containing the component
+/datum/component/two_handed/proc/on_attack_self(datum/source, mob/user)
+	if(wielded)
+		unwield(user)
+	else
+		wield(user)
+
+/**
+ * Wield the two handed item in both hands
+ *
+ * vars:
+ * * user The mob/living/carbon that is wielding the item
+ */
+/datum/component/two_handed/proc/wield(mob/living/carbon/user)
+	if(wielded)
+		return
+	if(ismonkey(user))
+		to_chat(user, "<span class='warning'>It's too heavy for you to wield fully.</span>")
+		return
+	if(user.get_inactive_held_item())
+		if(require_twohands)
+			to_chat(user, "<span class='notice'>[parent] is too cumbersome to carry in one hand!</span>")
+			user.dropItemToGround(parent, force=TRUE)
+		else
+			to_chat(user, "<span class='warning'>You need your other hand to be empty!</span>")
+		return
+	if(user.get_num_arms() < 2)
+		if(require_twohands)
+			user.dropItemToGround(parent, force=TRUE)
+		to_chat(user, "<span class='warning'>You don't have enough intact hands.</span>")
+		return
+
+	// wield update status
+	wielded = TRUE
+	RegisterSignal(user, COMSIG_MOB_SWAP_HANDS, .proc/on_swap_hands)
+	SEND_SIGNAL(parent, COMSIG_TWOHANDED_WIELD, user)
+	if(on_wield_callback)
+		on_wield_callback.Invoke(user)
```

Comment:
You don't need both the signal and callback, they have the same purpose. I gave them as alternatives to eachother

Reply:
I am now using signals no more callbacks, all items have been converted to them.